### PR TITLE
KAFKA-4306: Shutdown distributed herder with a timeout.

### PR DESCRIPTION
**Pull request by kkonstantine**
Resolves

KAFKA-4306: Connect workers won't shut down if brokers are not available
KAFKA-4154: Kafka Connect fails to shutdown if it has not completed startup

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java [hunk 1]
```diff
@@ -372,20 +379,17 @@ public void stop() {
...
-                // ignore, should not happen
-            }
-        }
-
-        forwardRequestExecutor.shutdown();
-        startAndStopExecutor.shutdown();
+        herderExecutor.shutdown();
         try {
-            if (!forwardRequestExecutor.awaitTermination(10000, TimeUnit.MILLISECONDS))
+            if (!forwardRequestExecutor.awaitTermination(workerTasksShutdownTimeoutMs, TimeUnit.MILLISECONDS))
```
Comment (shikhar) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java]: This should probably refer to `herderExecutor` :)

Comment (kkonstantine) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java]: yes!

**Comment by ewencp** [comment]
@kkonstantine I think something got messed up with the branch and some unrelated changes are getting included.

**Comment by ewencp** [comment]
Oh, nm, had this stale PR open instead of the one against the Kafka repo.

**Pull request closed without merging**